service/middleware.go

```diff
@@ -281,11 +288,6 @@ func createProxyRequestMiddleware(next http.Handler, config config.Config, servi
 					Msg("cache miss")
 
 				w.Header().Add(cachemdw.CacheHeaderKey, cachemdw.CacheMissHeaderValue)
-				// add CORS headers (if not already added)
-				accessControlAllowOriginValue := config.GetAccessControlAllowOriginValue(r.Host)
-				if w.Header().Get("Access-Control-Allow-Origin") == "" && accessControlAllowOriginValue != "" {
-					w.Header().Set("Access-Control-Allow-Origin", accessControlAllowOriginValue)
-				}
```

Comment:
this addition caused the CORs header to be set twice on live deployments (once here, then again from the backend). removing it gives full power to the backend to set all headers.

this is kept for cache hits because we want to ensure correct CORs headers are set for responses that get cached without them. tests are updated to only require CORs header on cache hit tests (locally they do not get the header, but in live deployments nginx sets it).